runtime/src/main/scala/com/soundcloud/twinagle/ServerBuilder.scala

```diff
@@ -0,0 +1,32 @@
+package com.soundcloud.twinagle
+
+import com.twitter.finagle.http.{Request, Response}
+import com.twitter.finagle.{Filter, Service}
+import com.twitter.util.Future
+import scalapb.{GeneratedMessage, GeneratedMessageCompanion, Message}
+
+case class ServerBuilder(
+    extension: EndpointMetadata => Filter.TypeAgnostic,
+    endpoints: Map[EndpointMetadata, Service[Request, Response]] = Map.empty
+) {
+
+  def register[
+      Req <: GeneratedMessage with Message[Req]: GeneratedMessageCompanion,
+      Resp <: GeneratedMessage with Message[Resp]: GeneratedMessageCompanion
+  ](endpointMetadata: EndpointMetadata, rpc: Req => Future[Resp]): ServerBuilder =
+    registerService(endpointMetadata, Service.mk(rpc))
+
+  def registerService[
```

Comment:
Is/will this being used anywhere (besides directly above)? If not, why not just make one method?

Reply:
good point, :+1: